lib/trino-filesystem-cache-alluxio/src/main/java/io/trino/filesystem/alluxio/TracingCacheManager.java

```diff
@@ -0,0 +1,162 @@
+                .startSpan();
+        return withTracing(span, () -> delegate.get(pageId, position, length, readTargetBuffer, cacheContext));
+    }
+
+    @Override
+    public int getAndLoad(PageId pageId, int position, int length, ReadTargetBuffer readTargetBuffer, CacheContext cacheContext, Supplier<byte[]> supplier)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.getAndLoad")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .setAttribute(CACHE_FILE_READ_POSITION, (long) position)
+                .setAttribute(CACHE_FILE_READ_SIZE, (long) length)
+                .startSpan();
+        return withTracing(span, () -> delegate.getAndLoad(pageId, position, length, readTargetBuffer, cacheContext, supplier));
+    }
+
+    @Override
+    public void deleteFile(String fileId)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.deleteFile")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .startSpan();
+        withTracing(span, () -> delegate.deleteFile(fileId));
+    }
+
+    @Override
+    public void deleteTempFile(String fileId)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.deleteTempFile")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .startSpan();
+        withTracing(span, () -> delegate.deleteTempFile(fileId));
+    }
+
+    @Override
+    public boolean delete(PageId pageId)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.delete")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .startSpan();
+        return withTracing(span, () -> delegate.delete(pageId));
+    }
+
+    @Override
+    public boolean append(PageId pageId, int position, byte[] bytes, CacheContext cacheContext)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.append")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .setAttribute(CACHE_FILE_WRITE_POSITION, (long) position)
+                .setAttribute(CACHE_FILE_WRITE_SIZE, (long) bytes.length)
+                .startSpan();
+        return withTracing(span, () -> delegate.append(pageId, position, bytes, cacheContext));
+    }
+
+    @Override
+    public void commitFile(String fileId)
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.commitFile")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .startSpan();
+        withTracing(span, () -> delegate.commitFile(fileId));
+    }
+
+    @Override
+    public Optional<DataFileChannel> getDataFileChannel(PageId pageId, int position, int length, CacheContext cacheContext)
+            throws PageNotFoundException
+    {
+        Span span = tracer.spanBuilder("AlluxioCacheManager.getDataFileChannel")
+                .setAttribute(CACHE_KEY, cacheKey)
+                .setAttribute(CACHE_FILE_WRITE_POSITION, (long) position)
+                .setAttribute(CACHE_FILE_WRITE_SIZE, (long) length)
```

Comment:
`getDataFileChannel` looks like a read operation, so we should use the read version of the attribute keys.

Reply:
Fixed